ovve_ui.py

```diff
@@ -10,48 +10,51 @@
 from PyQt5 import QtCore, QtGui, QtSerialPort, QtWidgets, uic
 from PyQt5.QtCore import *
 from PyQt5.QtGui import *
-from PyQt5.QtWidgets import (
-    QAbstractButton,
-    QApplication,
-    QHBoxLayout,
-    QLabel,
-    QPushButton,
-    QStackedWidget,
-    QVBoxLayout,
-    QWidget,
-)
+from PyQt5.QtWidgets import (QAbstractButton, QApplication, QHBoxLayout,
+                             QLabel, QPushButton, QStackedWidget, QVBoxLayout,
+                             QWidget)
 
 from params import Params
 from settings import Settings
+from ui_settings import (DisplayRectSettings, FancyButtonSettings,
+                         SimpleButtonSettings, TextSetting, UISettings)
 
 
+# TODO: Abstract text placement
 class FancyDisplayButton(QAbstractButton):
-    def __init__(self, label, value, unit, parent=None, size=(200, 100)):
+    def __init__(self,
+                 label,
+                 value,
+                 unit,
+                 button_settings,
+                 parent=None,
+                 size=(200, 100)):
         super().__init__(parent)
         self.label = label
         self.value = value
         self.unit = unit
         self.size = size
+        self.button_settings = button_settings
 
     def paintEvent(self, event):
         painter = QPainter(self)
 
-        labelFont = QFont("Times", 20, QFont.Bold)
-        numberFont = QFont("Times", 36, QFont.Bold)
-        unitFont = QFont("Times", 18)
+        labelFont = self.button_settings.labelFont
+        numberFont = self.button_settings.numberFont
+        unitFont = self.button_settings.unitFont
 
-        painter.setBrush(QBrush(QColor("#d2fcdc")))
+        painter.setBrush(self.button_settings.getFillBrush())
         painter.drawRect(0, 0, *self.size)
-        painter.setPen(QPen(QColor("#3ed5f0")))
+        painter.setPen(self.button_settings.getLabelPen())
         painter.setFont(labelFont)
         painter.drawText(int(self.size[0] / 2 - 50), int(self.size[1] / 5),
```

Comment:
I think you could create a function in button settings to calcuate the x,y position of the text.